kafkaconsumer/consumer_manager.go

```diff
@@ -0,0 +1,421 @@
+package kafkaconsumer
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/samuel/go-zookeeper/zk"
+	"github.com/wvanbergen/kazoo-go"
+	"gopkg.in/tomb.v1"
+)
+
+// Consumer represents a consumer instance and is the main interface to work with as a consumer
+// of this library.
+type Consumer interface {
+	// Interrups will initiate the shutdown procedure of the consumer, and return immediately.
+	// When you are done using the consumer, you must either call Close or Interrupt to prevent leaking memory.
+	Interrupt()
```

Comment:
Why bother introducing `Interrupt` when you can always do `go c.Stop()` why you want to stop asynchronously?